contracts/Router.sol

```diff
@@ -0,0 +1,133 @@
+        return _released[token][account];
+    }
+
+    function totalShares() public view returns (uint256) {
+        return _totalShares;
+    }
+
+    function totalReleased(address token) public view returns (uint256) {
+        return _totalReleased[token];
+    }
+    // ---------------------------- external functions ----------------------------
+    function release(IERC20 token) external nonReentrant {
+        require(_totalShares > 0, "Router: no shares");
+        uint256 balance = token.balanceOf(address(this));
+        uint256 total = 0;
+        for(uint256 i = 0; i < _payees.length; i++) {
+            address payee = _payees[i];
+            uint256 payment;
+            if (i == _payees.length - 1){
+                payment = balance - total;
+            } else {
+                payment = balance * _shares[payee] / _totalShares;
+            }
+            if (payment > 0) {
+                _released[payee][address(token)] = _released[payee][address(token)] + payment;
+                SafeERC20.safeTransfer(token, payee, payment);
+                total += payment;
+            }
+        }
+        _totalReleased[address(token)] = _totalReleased[address(token)] + total;
+        emit PaymentReleased(token, total);
+    }
+
+     function addPayee(address account, uint256 shares_) external onlyOwner {
+        _addPayee(account, shares_);
+    }
+
+    function removePayee(address account) external onlyOwner {
+        _removePayee(account);
+    }
+
+    function adjustShare(address account, uint256 shares_) external onlyOwner {
+        _adjustShare(account, shares_);
+    }
+
+
+    
+    // ---------------------------- private functions ----------------------------
+
+    function _addPayee(address account, uint256 shares_) private {
+        require(account != address(0), "Router: account is the zero address");
+        require(shares_ > 0, "Router: shares are 0");
+        require(_shares[account] == 0, "Router: account already has shares");
+
+        _payees.push(account);
+        _shares[account] = shares_;
+        _totalShares = _totalShares + shares_;
+        emit PayeeAdded(account, shares_);
+    }
+
+
+    function _removePayee(address account) private {
+        require(account != address(0), "Router: account is the zero address");
+        require(_shares[account] > 0, "Router: account has no shares");
+
+        for (uint256 i = 0; i < _payees.length; i++) {
+            if (_payees[i] == account) {
+                _payees[i] = _payees[_payees.length - 1];
+                _totalShares = _totalShares - _shares[account];
+                emit PayeeRemoved(account, _shares[account]);
+                _shares[account] = 0;
+                _payees.pop();
+                break;
+            }
+        }
+    }
+
+    function _adjustShare(address account, uint256 shares_) private {
+        require(account != address(0), "Router: account is the zero address");
```

Comment:
We're not currently testing this require statement fails when the account is the zero address. We should add a test where we attempt to adjust the shares for the zero address